kitty/complete.py

```diff
@@ -86,6 +94,11 @@ def bash_input_parser(data):
     return words, new_word
 
 
+@input_parser
+def fish_input_parser(data):
+    return data.rstrip().splitlines(), True
```

Comment:
This does not look right to me. new_word should be True only if the cursor is after a space.

For example:

kitty +kitt<cursor>

should have new_word False, while

kitty +kitten <cursor>

should have new_word True

Reply:
Thing is that completions work quite differently in Fish than in other command line shells. As Fish interprets and filters the completions suggestions, there's no need to work on completing the current partial token. Only if we're doing something really tricky with the partial token. In that sense, `kitty +complete fish` will always receive complete tokens.